docs/docs/dev/node_management.md

```diff
@@ -0,0 +1,96 @@
+# Node Management
+
+Bacalhau clusters are composed of requester nodes, and compute nodes. The requester nodes are responsible for managing the compute nodes that make up the cluster. This functionality is only currently available when using NATS for the network transport.
+
+The two main areas of functionality for the requester nodes are, managing the membership of compute nodes that require approval to take part in the cluster, and monitoring the health of the compute nodes. They are also responsible for collecting information provided by the compute nodes on a regular schedule.
+
+## Compute node membership
+
+As compute nodes start, they register their existence with the requester nodes. Once registered, they will maintain a sentinel file to note that they are already registered, this avoids unnecessary registration attempts.
```

Comment:
When authentication is enabled, in order to even connect to a requester node, compute nodes must have the correct secret/auth values, right?

Asking as I am curious about the behavior when two compute nodes with the same ID/Name attempt to register, if one is approved will the second node with the same ID also be considered approved? The ID is just a field users provide in their config file so its very possible for two nodes to have the same ID/Name